dataset/create_twitter_dataset.py

```diff
@@ -0,0 +1,144 @@
+# This file takes the twitter dataset and transform it into the
+# format described in https://github.com/paivaspol/EECS591/issues/25
+# The result dataset will support data interdependency.
+import argparse
+import math
+import random
+import uuid
+
+# Constants for twitter modes.
+TWITTER_RT = 'RT'
+TWITTER_MT = 'MT'
+TWITTER_RE = 'RE'
+UUID_LIST_FILE = 'uuid_list.txt'
+USER_IP_MAP_FILE = 'user_ip_map.txt'
+
+# Generates the dataset.
+#
+# params:
+#   filename: the name of the file containing the twitter data
+#   user_ip_map: a map containing the user
+def generate_dataset(filename, user_ip_map, uuid_server_map):
+    print 'Generating access logs...'
+    with open(filename, 'rb') as dataset, open(filename + '.ready', 'wb') as result_file, open(UUID_LIST_FILE, 'wb') as uuid_file:
+        for line in dataset:
+            line_trimmed = line.strip()
+            line_splitted = line.split(' ')
+            timestamp = line_splitted[2]
+            mode = line_splitted[3].strip()
+            first_uid = line_splitted[0]
+            first_user_ip = user_ip_map[first_uid]
+            second_uid = line_splitted[1]
+            second_user_ip = user_ip_map[second_uid]
+            target_server = uuid_server_map[first_uid]
+            tweet_uid = str(uuid.uuid4())
+            # Schema: [timestamp]\t[uuid]\t[src]\t[dependency]\t[dest]\t[type]\t[status]\t[size]
+            if mode == TWITTER_MT:
+                # Three steps:
+                #   1) A reads A's timeline
+                #   2) A writes to his own timeline
+                #   3) A reads his own timeline
+                result_file.write('%s\t%s\t%s\t%s\t%s\t%s\t%s\n' % (timestamp, first_uid, first_user_ip, 'null', target_server, 'READ', '2800'))
```

Comment:
I see what you're saying. Hmm. I'm wondering if it would be helpful for our test logs to be the exact same format as the server generated formats? That way it's easy to replay a specific log? Whatcha think